include/nodes/internal/DataModelRegistry.hpp

```diff
@@ -7,9 +7,21 @@
 #include <QtCore/QString>
 
 #include "NodeDataModel.hpp"
+#include "Converter.hpp"
 #include "Export.hpp"
 #include "QStringStdHash.hpp"
 
+
+namespace std {
+
+  inline
+  bool operator<(QtNodes::NodeDataType const & d1,
+                 QtNodes::NodeDataType const & d2)
+  {
+    return d1.id < d2.id;
+  }
+}
+
```

Comment:
This may be the cause of [this compilation error](https://travis-ci.org/paceholder/nodeeditor/jobs/362367315#L1053):

```
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.1/../../../../include/c++/5.4.1/bits/stl_pair.h:221:24: error: 
      call to function 'operator<' that is neither visible in the template
      definition nor found by argument-dependent lookup
    { return __x.first < __y.first
                       ^
```

Firstly, **it's not allowed to add anything in `namespace std`**, unless it's explicitly stated that you can (e.g. specializations of `std::hash`).

Second, this is unnecessary. You can add the `operator<` to the `QtNodes` namespace, and [Argument Dependent Lookup (ADL)](http://en.cppreference.com/w/cpp/language/adl) will find it. In other words, just add this `operator<` following the declaration of `NodeDataType`, then `dataType1 < dataType2` will find the correct function.

Reply:
> Firstly, it's not allowed to add anything in namespace std

I used a lot `std::unordered_map` at work and blindly wrote the `operator<()` as if I were implementing the `hash` function. I'll fix that.